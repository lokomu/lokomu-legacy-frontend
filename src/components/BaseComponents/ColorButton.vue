<template>
  <button
    type="button"
    class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center"
    :class="[color, focusRingColor]"
  >
    {{ text }}
  </button>
</template>

<script setup lang="ts">
type ButtonColor = "red" | "green" | "blue";

const props = withDefaults(
  defineProps<{
    text: string;
    buttonColor?: ButtonColor;
  }>(),
  {
    buttonColor: "blue",
  },
);

const colors = {
  red: "from-pink-500 via-pink-600 to-pink-700",
  green: "from-green-400 via-green-600 to-green-700",
  blue: "from-floyd-400 via-floyd-600 to-floyd-700",
};
const ringColors = {
  red: "focus:ring-pink-300 dark:focus:ring-pink-800",
  green: "focus:ring-green-300 dark:focus:ring-green-800",
  blue: "focus:ring-floyd-300 dark:focus:ring-floyd-800",
};

const color = colors[props.buttonColor];
const focusRingColor = ringColors[props.buttonColor];
</script>
