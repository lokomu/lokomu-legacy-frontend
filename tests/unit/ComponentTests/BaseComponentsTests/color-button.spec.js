import { mount } from '@vue/test-utils';
import ColorButton from '@/components/BaseComponents/ColorButton';
import {describe, it, expect} from "vitest";

describe('ColorButton', () => {
  const wrapper = mount(ColorButton, {
    props: {
      text: 'Test Button',
    },
  });

  it('renders with default props', () => {
    expect(wrapper.find('button').text()).toBe('Test Button');
    expect(wrapper.classes()).toContain('from-floyd-400');
    expect(wrapper.classes()).toContain('via-floyd-600');
    expect(wrapper.classes()).toContain('to-floyd-700');
    expect(wrapper.classes()).toContain('focus:ring-floyd-300');
  });

  it('renders with custom button color', () => {
    const wrapper = mount(ColorButton, {
      props: {
        text: 'Test Button',
        buttonColor: 'red',
      },
    });
    expect(wrapper.classes()).toContain('from-pink-500');
    expect(wrapper.classes()).toContain('via-pink-600');
    expect(wrapper.classes()).toContain('to-pink-700');
    expect(wrapper.classes()).toContain('focus:ring-pink-300');
  });

  it('emits click event when clicked', async () => {
    await wrapper.trigger('click');
    expect(wrapper.emitted().click).toBeTruthy();
  });
  
  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});