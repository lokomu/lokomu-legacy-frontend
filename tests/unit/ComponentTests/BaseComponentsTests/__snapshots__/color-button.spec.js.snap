// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorButton > renders correctly 1`] = `
<button
  class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center from-floyd-400 via-floyd-600 to-floyd-700 focus:ring-floyd-300 dark:focus:ring-floyd-800"
  type="button"
>
  Test Button
</button>
`;
